src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
```

Comment:
If we are going to implement a singleton then the utility methods below should not be static (otherwise the singleton wouldn't have a purpose). Also would it be possible to use annotations instead of implementing manually the singleton logic?